docker_practice/api_practice.py

```diff
@@ -1,77 +1,128 @@
-
-    response = requests.post(url=commit_container_url, params=request_paras, headers=request_header)
-    print(response.status_code)
-    if response.status_code == 201:
-        print("commit successful")
-    else:
-        print("commit fail: ", response.text)
-
-#将commit提交到repo上
-def push_commit(url, registry_auth, repo_name):
-    push_commit_url = url + "/images/" + repo_name + "/push"
-    request_header = {"X-Registry-Auth": registry_auth}
-
-    response = requests.post(push_commit_url, headers=request_header)
-    print(push_commit_url)
-    if response.status_code == 200:
-        print("push successful")
-    else:
-        print("push fail: ", response.text)
-
-
-
-if __name__ == '__main__':
-    url = "http://127.0.0.1:2376"
+DAEMON_URL = "http://127.0.0.1:2376"
+def check_connection(daemon_url=DAEMON_URL):
+    """
+    Test the docker connection
+
+    :param daemon_url: docker daemon url
+    :return: the first parameter is equals to 200 or not, the second parameter is the response body
+    """
+    check_connection_url = os.path.join(daemon_url, "containers/json")
+    params = {"all": 1}
+    r = requests.get(url=check_connection_url, params=params)
+    return r.status_code == 200, r.text
+
+
+def create_container(
+        container_name, *,
+        daemon_url=DAEMON_URL,
+        image="ubuntu"):
+    """
+    Create a new container
+
+    :param container_name:  container name
+    :param daemon_url: docker daemon url
+    :param image: name of the image which you create the container based on
+    :return: the first parameter is equals to 200 or not, the second parameter is the response body
+    """
+
+    r = requests.post(
+        url=os.path.join(daemon_url, "containers/create"),
+        params={"name": container_name},
+        json={"Image": image},
+        headers={"Content-Type": "application/json"})
+    return r.status_code == 201, r.text
+
+
+def commit_container(
+        container_name, *,
+        daemon_url=DAEMON_URL,
+        comment="my comments"):
+    """
+    Commit the changes to container
+
+    :param container_name: container name
+    :param daemon_url: docker daemon url
+    :param comment: the comment of the commit
+    :return: the first parameter is equals to 200 or not, the second parameter is the response body
+    """
+
+    r = requests.post(
+        url=os.path.join(daemon_url, "commit"),
+        params={
+            "container": container_name,
+            "comment": comment
+        },
+        headers={"Content-Type": "application/json"})
+    return r.status_code == 201, r.text
+
+
+def push_modified_commit(
+        repo_name="palanqu/docker_learning", *,
+        daemon_url=DAEMON_URL,
+        registry_auth):
+    """
+    Push the commit to repo
+    :param repo_name: repo name
+    :param daemon_url: docker daemon url
+    :param registry_auth: the registry_auth of the user, you need commit the json like {"username":"","password":"", "auth":"","email":""} to base64
```

Comment:
This line is too long.

Consider using automated tools such as `yapf` and `pep8` to auto-reformat python code to conform to the python coding standards.